### DISCUSSION TITLE
Test result fails

### DISCUSSION THREAD

**Discussion by yiping-tks** [nkicsl/Resfusion#17] [General]
Hi, thanks for your opening source of such a great job! 
Recently, I test your code on my own dataset (a few hundreds of images) for image restoration task. I found that the model can fastly converged in the training data and the restored results is almost perfect. However, the model fails on the test set, showing blurry results, during which the test psnr will increase to a relatively worse value and then degrade. 
![image](https://github.com/user-attachments/assets/973e87ed-7d8c-4409-be66-e6846e4758f6)
I trained the model from scratch with an old version of pytorch and do not use pl lightning, I wonder is there anything I should care about. Or could you please give me some insight to handle this problem~
Thanks again, wish you a good day~

**Comment by YukinoshitaLove** [comment]
@Jotaro-X  From the pictures you provided, the PSNR of the training set kept increasing, while the PSNR of the validation set first increased and then decreased. The PSNR of the validation set is much better than the initial PSNR, which proves that `Resfusion` has indeed learned some patterns. The subsequent decrease of PSNR is usually due to overfitting. Since your dataset may be relatively small, it may be a good choice to try a smaller model or fine-tune a pretrained model. As for the PSNR of the validation set being lower than that of the training set, this is a normal phenomenon, because there is usually a difference in the distribution of the two (especially when the dataset is not large). `ฅ^•ﻌ•^ฅ`

**Reply by yiping-tks** [reply]
tks for your quick reply~
Yes, the problem is that the model quickly overfit on my dataset which is relatively small. I introduced some data augmentation and regularization to resolve this problem. 
By the way, I find the datasets in the paper is also not so big (compared to stable diffusion trained on billions of image pairs), consisting of hundreds image pairs. If I train your model on large scale datasets for image restoration tasks, perhaps with thousands of or more than ten thousands of images, do you think your model will still work well ?~ or some tips when training on such a big dataset~
hope you a good day~

**Reply by yiping-tks** [reply]
oh, I forget one question~Is there any tips to increase the resolution of the output image? I tried your model on larger res (512), and the consumption is large.

**Reply by YukinoshitaLove** [reply]
I think it is possible. For the image translation task, I used about 50K images for training. The results show that Resfusion can learn effective patterns. For the second question, we are working on it, please stay tuned for our latest work (will be released soon).

**Reply by yiping-tks** [reply]
oh, looking forward to your further jobs！I also find that in your code, you provide some other choices for some modules. Such as for denoising module, you provide RDDM_Unet, DDIM_Unet and DiT. Have u ever compared their differences on different tasks~I find your default choice is RDDM_Unet and the pre-trained ckpt (raindrop) is trained based on it.

**Reply by YukinoshitaLove** [reply]
DIT shows scalability, but Unet wroks better with small datasets. DDIM_UNET is mainly used for fair comparison with other generative models.

**Reply by yiping-tks** [reply]
I see! tks a lot~

**Reply by YukinoshitaLove** [reply]
@yiping-tks 

> oh, I forget one question~Is there any tips to increase the resolution of the output image? I tried your model on larger res (512), and the consumption is large.

Our latest work uses some of the most cutting-edge technical advances to scale the residual diffusion model up to the 1-billion-parameter level, and in theory it can support any high resolution that pretrained models can handle. The results are truly impressive. Feel free to star our project: https://github.com/YukinoshitaLove/ScaleResfusion.

**Comment by yiping-tks** [comment]
Copy that!

**Comment by yiping-tks** [comment]
Copy that!
